src/base/reduce.jl

```diff
@@ -9,21 +9,27 @@ reduce1arg = [
 (:maxabs_,  :((y.==abs.(x)).*sign.(x))),
 (:minabs_,  :((y.==abs.(x)).*sign.(x))),
 ]
-sumabs_(x...)=sum(abs,x...)
-sumabs2_(x...)=sum(abs2,x...)
-minabs_(x...)=minimum(abs,x...)
-maxabs_(x...)=maximum(abs,x...)
+sumabs_(x...; kargs...)=sum(abs,x...; kargs...)
+sumabs2_(x...; kargs...)=sum(abs2,x...; kargs...)
+minabs_(x...; kargs...)=minimum(abs,x...; kargs...)
+maxabs_(x...; kargs...)=maximum(abs,x...; kargs...)
 
 _ones(x::Rec{T}) where T<:Number = one(T) #fix #56
-_ones(x::Rec) = ones(x)
+_ones(x::Rec) = fill(1.0, size(x.value))
```

Comment:
This edit breaks the type safety of AutoGrad functions.  You always create Array{Float64, N} whatever the type of x.value . If I use `ones(x)`, both Knet and AutoGrad pass tests.